### ISSUE TITLE
Device ports over 9 are displayed differently

### ISSUE THREAD

**Issue by ovenwerks** [Houston4444/HoustonPatchbay#2]
See image attached. Ports 0 through 9 are displayed as playback_AUX with port number in boxes and ports over 9 are displayed as playback with AUX10, AUX11 in little boxes.

Note this is with pipewire and may not show up with jack that has no AUX all over the place. Also in PW where ports have the same name (like input_FL) the second similar named port will have a number appended to it. For example: Plasma PA:input_FL-367. These also display differently.

![patchanceover9](https://user-images.githubusercontent.com/12753609/204628892-fbe5ca17-3b0c-4982-b13e-39353e607a8c.png)

**Comment by Houston4444** [comment]
This issue has already been fixed. Maybe you use RaySession or Patchance last release, but not git master. See https://github.com/Houston4444/RaySession/issues/147

**Comment by ovenwerks** [comment]
Thank you, does that fix also deal with what is seen in this image?
![LR_image](https://user-images.githubusercontent.com/12753609/204839224-72afbf57-8834-466d-87f5-554fa9fd7778.png)

**Comment by Houston4444** [comment]
No, it wont.
Stereo detection is not perfect, and it can't be. It's based on port names, checking if a port can be grouped with the previous one.
Here, admit that port names are really strange, jumping 2 values from the left to the right one (FL-367, FR-369...). Anyway it is not understandable for normal user to know what port names means !

We have some solutions:
* do nothing, and let user group the ports (right click on a port -> set as stereo with)
* Implement a new rule, but what a strange rule ! risk to group wrong ports together exists.
* Ask to the concerned devs why theses names are written this way, and convince them to change.

**Comment by ovenwerks** [comment]
"Ask the devs to set portgroup metadatas on the ports" This is probably the best solution.

Feel free to call this closed.

As a new user of pipewire, I am struggling to come to terms with the mixture of desktop audio I am really not interested in and jack audio which I am interested in. I am at least able to hide all the "Monitor" ports and use shorter names or change the names of devices. Thank you for the time.

**Comment by Houston4444** [comment]
I agree, metadata portgroups would be the best solution (change the names could be nice too). That said, I don't know where to call ask for a change... 

> I am at least able to hide all the "Monitor" ports and use shorter names or change the names of devices.

Do you plan to make a GUI to control pipewire settings ? or adapt studio-controls to pipewire ?

**Comment by ovenwerks** [comment]
On Thu, 1 Dec 2022, Houston4444 wrote:

> Do you plan to make a GUI to control pipewire settings ? or adapt
> studio-controls to pipewire ?

Studio-controls is that GUI ;)  I have it set to switch between 
pulse/jack, pipewire, pipewire/jack and pipewire with jack as a device. So 
far I have the switch between pa/jack to pipewire and back working pretty 
good. I have a tab to set PW settings and my device tab will change device 
names in PW. My finger in the wind says most people will be using pipewire 
by default very soon and I had better add it sooner than later. pulse is 
going away even for those who want to use jack. It is too bad that A) 
pipewire does not use jack backends like ffado or B) the ALSA firewire 
modules are so poor. That is what has kept me from working on pipewire 
sooner. I have not yet gotten a PW jack bridge running but I think that is 
just a matter of time, the code is there. The reality is that I need to 
have pipewire as a replacement for pa/jack working first so that is what I 
am working on.

**Comment by Houston4444** [comment]
Cool ! I will give a try.

> It is too bad that A) pipewire does not use jack backends like ffado

I am in this case, so I may still use JACK for a long time.
